src/Compilers/CSharp/Portable/Symbols/Synthesized/ReadOnlyListType/SynthesizedReadOnlyListTypeSymbol.cs

```diff
@@ -0,0 +1,614 @@
+
+            TypeSymbol fieldType = hasKnownLength ?
+                compilation.CreateArrayTypeSymbol(elementType: typeParameter) :
+                compilation.GetWellKnownType(WellKnownType.System_Collections_Generic_List_T).Construct(typeArgs);
+            _field = new SynthesizedFieldSymbol(this, fieldType, "_items", isReadOnly: true);
+
+            var iEnumerable = compilation.GetSpecialType(SpecialType.System_Collections_IEnumerable);
+            var iEnumerableT = compilation.GetSpecialType(SpecialType.System_Collections_Generic_IEnumerable_T).Construct(typeArgs);
+            var iReadOnlyCollectionT = compilation.GetSpecialType(SpecialType.System_Collections_Generic_IReadOnlyCollection_T).Construct(typeArgs);
+            var iReadOnlyListT = compilation.GetSpecialType(SpecialType.System_Collections_Generic_IReadOnlyList_T).Construct(typeArgs);
+            var iCollectionT = compilation.GetSpecialType(SpecialType.System_Collections_Generic_ICollection_T).Construct(typeArgs);
+            var iListT = compilation.GetSpecialType(SpecialType.System_Collections_Generic_IList_T).Construct(typeArgs);
+
+            _interfaces = ImmutableArray.Create(
+                iEnumerable,
+                iEnumerableT,
+                iReadOnlyCollectionT,
+                iReadOnlyListT,
+                iCollectionT,
+                iListT);
+
+            var membersBuilder = ArrayBuilder<Symbol>.GetInstance();
+            membersBuilder.Add(
+                _field);
+            membersBuilder.Add(
+                new SynthesizedReadOnlyListConstructor(this, fieldType));
+            membersBuilder.Add(
+                new SynthesizedReadOnlyListMethod(
+                    this,
+                    (MethodSymbol)compilation.GetSpecialTypeMember(SpecialMember.System_Collections_IEnumerable__GetEnumerator),
+                    generateGetEnumerator));
+            membersBuilder.Add(
+                new SynthesizedReadOnlyListMethod(
+                    this,
+                    ((MethodSymbol)compilation.GetSpecialTypeMember(SpecialMember.System_Collections_Generic_IEnumerable_T__GetEnumerator)!).AsMember(iEnumerableT),
+                    generateGetEnumeratorT));
+            addProperty(membersBuilder,
+                new SynthesizedReadOnlyListProperty(
+                    this,
+                    ((PropertySymbol)compilation.GetWellKnownTypeMember(WellKnownMember.System_Collections_Generic_IReadOnlyCollection_T__Count)!).AsMember(iReadOnlyCollectionT),
+                    generateCount));
+            addProperty(membersBuilder,
+                new SynthesizedReadOnlyListProperty(
+                    this,
+                    ((PropertySymbol)compilation.GetWellKnownTypeMember(WellKnownMember.System_Collections_Generic_IReadOnlyList_T__Item)!).AsMember(iReadOnlyListT),
+                    generateIndexer));
+            addProperty(membersBuilder,
```

Comment:
I'd considered having a single property implement members from multiple interfaces but I don't know if we do that elsewhere so it might increase complexity overall.